pkg/apis/eventing/v1alpha1/eventtype_types.go

```diff
@@ -81,3 +82,13 @@ type EventTypeList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []EventType `json:"items"`
 }
+
+// GetGroupVersionKind returns GroupVersionKind for EventType
```

Comment:
Could we add a check that all these types actually implement what we expect them to implement. Something similar to L45-L50.

Reply:
Done now. Added validations for implementing `OwnerRefable` (where `GetGroupVersionKind` is defined) an also `HasSpec` (where `GetUntypedSpec` is defined)